### PR TITLE
refactor: using croaring instead of roaring-rs

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Using croaring instead of roaring-rs, with this PR, the performance is not related to roaring bitmap implementation.

After [this PR](https://github.com/RoaringBitmap/croaring-rs/pull/112) is merged, I'll replace the patched crate with the official crate.

- Closes #issue

### CODE REVIEW

**Comment by Xuanwo** [comment]
> Using croaring instead of roaring-rs.

Hi, is there context for this change? Using a pure rust lib seems a better idea.

**Comment by ariesdevil** [comment]
> Hi, is there context for this change? Using a pure rust lib seems a better idea.↳

The pure rust implementation is just a reference implementation, the performance gap is large, and it's no longer being maintained, only maintaining the c binding implementation.

**Comment by ariesdevil** [comment]
@everpcpc I can build the python binding success on my local linux machine, do you have any ideas of the failed build in CI environment?

File: src/query/functions/src/scalars/bitmap.rs [hunk 1]
```diff
@@ -48,8 +49,8 @@ pub fn register(registry: &mut FunctionRegistry) {
                     result
                 }) {
                 Ok(v) => {
-                    let rb = RoaringTreemap::from_iter(v.iter());
-                    rb.serialize_into(&mut builder.data).unwrap();
+                    let rb = Treemap::from_iter(v);
+                    builder.data = rb.serialize().unwrap();
```
Comment (b41sh) [inline src/query/functions/src/scalars/bitmap.rs:53]: value should be pushed into the builder

**Comment by everpcpc** [comment]
> @everpcpc I can build the python binding success on my local linux machine, do you have any ideas of the failed build in CI environment?

Seems like the gcc is too old in python build image?

**Comment by b41sh** [comment]
@ariesdevil almost ok for me, please fix the ci

**Comment by ariesdevil** [comment]
The failed python binding caused by [this reason](https://github.com/PyO3/maturin-action/issues/197)

**Review by sundy-li — APPROVED** [review]

**Review by b41sh — APPROVED** [review]

**Comment by ariesdevil** [comment]
Many thanks @everpcpc for helping ci pass.

**Comment by BohuTANG** [comment]
This breaking the env in some places, and we are considering reverting it, cc @ariesdevil

**Pull request merged**